DS3231.cpp

```diff
@@ -180,6 +181,57 @@ DateTime RTClib::now(TwoWire & _Wire) {
   return DateTime (y, m, d, hh, mm, ss);
 }
 
+//Subtraction returns difference in seconds between 2 DateTimes.
+//This is signed so that you can express situations when the LHS is earlier in time than the RHS.
+long DateTime::operator - (DateTime const &rhs){
+	return this->unixtime() - rhs.unixtime();
+}
+
+//Addition accepts seconds as the RHS, RHS >= 0
+//and returns a new DateTime where RHS is added to LHS.
+//I did not implement addition of negative numbers because that causes underflow with the
+//unixtime uint32_t.
+DateTime DateTime::operator + (uint32_t const &rhs){
+	//guard against overflow. if this->unixtime() + rhs > UINT32_MAX, just return the maximum value
+	if (UINT32_MAX - rhs < this->unixtime()){
+		return UINT32_MAX;
```

Comment:
I think returning the MAX value here is misleading in a way.
I would recommend here normally to return an error, but this might be difficult here.
Thus I would return `DataTime()` ?> 0.